### DISCUSSION TITLE
The protocol stack calls the emitter to enter the dead loop

### DISCUSSION THREAD

**Discussion by tuopu** [openthread/openthread#8104] [Q&A] [answered]
The protocol stack calls the function "otplatradioreceive". Since the radio is in the transmitting state, it returns ot_ ERROR_ INVALID_ State, enter the dead loop. Please ask about the logical sequence of calling the radio, and how to avoid calling the otplatradioreceive function in the transmitting state.

**Comment by abtink** [comment, marked as answer]
Calling `otPlatRadioReceive` is valid during tx and is expected to be handled by radio platform implementation. 
The expected behavior is that platform must abort/stop the oging tx and enter rx mode. For example, this can be used if/when "ack timeout" is implemented by OT `SubMac` code.

Here are the valid radio transitions (from [`platform/radio.h`](https://github.com/openthread/openthread/blob/main/include/openthread/platform/radio.h)):
```
/**
 * The following are valid radio state transitions:
 *
 *                                    (Radio ON)
 *  +----------+  Enable()  +-------+  Receive() +---------+   Transmit()  +----------+
 *  |          |----------->|       |----------->|         |-------------->|          |
 *  | Disabled |            | Sleep |            | Receive |               | Transmit |
 *  |          |<-----------|       |<-----------|         |<--------------|          |
 *  +----------+  Disable() +-------+   Sleep()  +---------+   Receive()   +----------+
 *                                    (Radio OFF)                 or
 *                                                        signal TransmitDone
 *
 * During the IEEE 802.15.4 data request command the transition Sleep->Receive->Transmit
 * can be shortened to direct transition from Sleep to Transmit if the platform supports
 * the OT_RADIO_CAPS_SLEEP_TO_TX capability.
 */
```

**Reply by tuopu** [reply]
thank you,After testing, it has indeed been realized. Thank you very much。
